### PR TITLE
ipi-deprovision: refactor to add gcp

### PR DESCRIPTION
**Pull request by stevekuznetsov**
Signed-off-by: Steve Kuznetsov <skuznets@redhat.com>

/cc @wki

### CODE REVIEW

File: cmd/ipi-deprovision/ipi-deprovision.sh [hunk 1]
```diff
@@ -4,64 +4,60 @@ set -o errexit
...
-  local cluster_name
-  cluster_name=$1
-  ###eg cluster_name=kubernetes.io/cluster/ci-op-724qy8fn-55c01-s5c8w
-  local ns
-  ns=${cluster_name:22:14}
-  echo "cluster is used by namespace ${ns}"
-  if ! oc get ns "${ns}" >/dev/null 2>&1; then
-    echo "namespace ${ns} does not exist any more"
-    return
+echo "deprovisioning in GCE ..."
```
Comment (wking) [inline cmd/ipi-deprovision/ipi-deprovision.sh]: Running GCE after AWS will expose us to GCE leaks if AWS gets stuck.  Compare openshift/ci-tools#231 where we worked around AWS-side teardown errors blocking us from removing other leaked AWS clusters.  I'd rather run the AWS and GCP portions in parallel (in separate scripts?  Based on a argument passed to the script?  In parallel in the one script via backgrounded functions?).

But as long as we don't get stuck on anything, serial is still better than "no GCP reaping at all".

Comment (wking) [reply cmd/ipi-deprovision/ipi-deprovision.sh]: Fixed via #387.

File: cmd/ipi-deprovision/ipi-deprovision.sh [hunk 2]
```diff
@@ -4,64 +4,60 @@ set -o errexit
...
-do
-  echo "doing region ${r} ..."
-  json_output=$(aws ec2 describe-vpcs --output json --region "${r}")
-  for cluster in $(echo ${json_output} | jq --arg date "${cluster_age_cutoff}" -r -S '.Vpcs[] | select (.Tags[]? | (.Key == "expirationDate" and .Value < $date)) | .Tags[] | select (.Value == "owned") | .Key' | shuf)
-  do
-    expirationDateValue=$(echo ${json_output} | jq --arg cl "${cluster}" -r -S '.Vpcs[] | select (.Tags[]? | (.Key == $cl and .Value == "owned")) | .Tags[] | select (.Key == "expirationDate") | .Value')
-    handle_cluster "${cluster}" "${expirationDateValue}" "${r}"
-    collect_metadata "${cluster}"
-  done
+for workdir in $( find /tmp/deprovision -mindepth 1 -type d ); do
```
Comment (jstuever) [inline cmd/ipi-deprovision/ipi-deprovision.sh]: extending what @wking  has said.... shuf here? `$( find /tmp/deprovision -mindepth 1 -type d | shuf )`

**Pull request merged**